Makefile

```diff
@@ -125,19 +125,26 @@ lint-fix: golangci-kal ## Run golangci-lint linter and perform fixes
 ##@ Build
 
 .PHONY: build
-build: manifests generate fmt vet ## Build manager binary.
-	go build -o bin/manager cmd/manager/main.go
+build: manifests generate fmt vet build-manager
+
+.PHONY: build-manager
+# Set build time variables including version details
+build-manager: LDFLAGS ?= $(shell source ./hack/version.sh; version::get_git_vars; version::get_build_date; version::ldflags)
+build-manager:
+	go build -ldflags "${LDFLAGS}" -o bin/manager cmd/manager/main.go
 
 .PHONY: run
 run: manifests generate fmt vet ## Run a controller from your host.
 	go run ./cmd/manager/main.go
 
+DOCKER_BUILD_ARGS ?=
+
 # If you wish to build the manager image targeting other platforms you can use the --platform flag.
 # (i.e. docker build --platform linux/arm64). However, you must enable docker buildKit for it.
 # More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
 docker-build: ## Build docker image with the manager.
-	$(CONTAINER_TOOL) build -t ${IMG} .
+	$(CONTAINER_TOOL) build -t ${IMG} ${DOCKER_BUILD_ARGS} .
```

Comment:
We should inject the build args in the `docker-build` target as well. I'll push a fix shortly.

Reply:
Scratch that. Because of the ARG directives, all variables are set to empty and the `version.sh` script is not updating them.

Reply:
The problem was that `version::ldflags` didn't call `version::get_git_vars` or `version::get_build_date` to set default values. It should be fixed now.